First Aid and Emergency Oxygen Tweaks

Novacat opened the pull request:
:cl: Novacat
   tweak: Advanced First Aid Kit: Swaps inaprovaline autoinjector for assorted pill bottle.
   tweak: Burn First Aid Kit: Adds three ointment kits, swaps inaprovaline autoinjector for tramadol autoinjector
   tweak: Oxygen first aid kits now contain inaprovaline pill bottle, dexalin pill bottle, and four inaprovaline autoinjectors
   tweak: Toxin first aid kits now contain a dylovene pill bottle, three hyronalin pills, and four autoinjectors of dylovene
   tweak: Autoinjectors are now the same size as syringes
   tweak: Emergency Oxygen/Nitrogen Tanks now start off full
   tweak: Regular Nitrogen tank starts off at the same level as oxygen tanks
   tweak: Paracemetol's painkiller effect increased from 25 to 35
   /:cl:

PurpleMartinJCK commented:
This change adds a lot of medication to the medkits in general. What stops people from nicking out pill bottles and carrying around an apothecary?  Why do they have pills and autoinjectors?

Novacat commented:
First aid kits have always contained pills, the only new additions are inaprovaline, tramadol, and hyronalin, and a lot of this is to either handle large numbers of casualties, or long term stabilization if someone cannot be treated by medical. Hyronalin was added to give the Torch (and Medical) a way to deal with radiation in rounds without a chemist, and Tramadol was added to adapt first aid kits to baymed where shock can send someone into cardiac. The autoinjectors are for immediate first aid treatment, and have the benefit that they can be used through voidsuits.

EcklesFire commented:
This is a huge buff for any non-med person who finds one of these easily scrounged kits, given how common all of them are in various first aid lockers.

Additionally I do not feel the advanced first aid kit should change. It's a balanced first aid kit that's mainly used by medical players as their primary source for their kits. Removing their burn kits is going to result in aux medbay looting like the lack of them used to.

The buff to burn kits is extreme, regular people grabbing emergency kits should not be having easy and ready access to burn kits.

Oxygen kits/shrug.

Toxin first aid kits are basically turned into a walking anti-bad guy with fancy gear kit, allowing you to easily survive weaponry that inflicts rads, like the phoron pistol, and allows anyone on board to basically ignore a supermatter delamination if they merely rush to the neareast first aid cabinet (and there's quite a few). 

Autoinjectors: Being able to pop off an easy 10u of drugs to anyone with these is a huge buff, and reinforces a fire and forget and carry on going self medicating culture that some people try to pull as non-medical. (You can basically turn yourself into an E-gun damage tank with two of these kits).

All in all a pretty unbalanced change that allows anyone who loots a wall cabinet to bypass medical for most things except surgical fixes and bloodloss. Especially given you can pop a bit of the ton of tramadol, do the high end patch of all your external wounds then hit yourself with 3 inap and 3 dylo auto injectors and fill yourself up to the gills with tricordrazine.

I do not believe the entire crew, or more specifically, the ones going looking for trouble, should be able to turn themselves into a walking healing bin. This is a crazy good buff for just about anyone who feels inclined to loot a first aid cabinet.

I have no problem with the oxygen tank buff though, the ones in survival kits are generally useless.

ghost commented:
The point of first-aid cabinets is to provide medication and tools to stabilize yourself, not to completely cure you of your ailments. For that purpose, the medical bay exists.

PurpleMartinJCK commented:
The crux of this for me is the question "what is the purpose of a med kit?". Are they a kit intended to be used to fix one human with one type of injury? Are they to provide enough meds for long term care? Are they intended to stabilise a human with a type of injury until medical gets there?

comma commented:
They are to provide tools to stave off death until someone qualified and equipped get there.
Patch bleeder, get some inapro in, some over the counter painkillers.
Whole raison'd'etre of medical is their monopoly on stuff for actual healing, by handing out that like free candy you take away a lot of existance from them. Skills etc don't matter, only gear does.

>First aid kits have always contained pills

That is a preeeetty new addition, unless you mean actual loose pills then yes. Pill bottles were added fairly recently and without a deep thought, just for sake of fitting the stuff like paracetamol in.

Also tramadol is a GAME CHANGER for any combat related situation, access to it should not be THAT easy. I can MAYBE see a single injector in a burn kits since those are extreme pain ones, but this brings to another point:

10u injectors are bad, injectors are extremely OP and doubling amout of chems you can effortlessly stick into people with zero delay is bad regardless of benign applications of that. 10u of Tramadol turns you into a laser soak for a pretty damn long time, that's no bueno either.
Not to mention that size change will wreck some loadouts and possibly boxes.

On lesser things: imo inapro pills are silly, inapro is more of emergency med, but then again its role changed quite a lot, maybe pills make more sense snow. I'd still go for syringes because 1) no delay on chem application, you want inapro to start NOW usually, and minor 2) less silly ICly to feed unconscious people things.

No strong feelings bout big oxygen tanks, thos are near infinite in my experience. Emergency tanks are kinda meant to be short, I dislike this change, but no strong feeling either, just my opinion.

tl;dr hard nope on opiate epidemic and 10u autoinjectors, oxy kit ok, tox kit ok if rad pills are scarce (1 or 2 loose pills instead of bottle), inapro should be in syringes IMO, burn kit MIGHT have 1 tramadol injector, but otherwise nope on those in kits, burn kits are important against infections/laser wounds, keep them.

Dukica99 commented:
**Ramble alert, I'm tired**

Well, overall, currently people barely if ever use any of the kits except the advanced kits, which are only used for looting burn and trauma kits. They aren't used as something you carry with you or pop out, just as a loot storage. Having them be useful would be great, but then the question moves towards people abusing them. I honestly think it's dependent on two things, actually medically trained people using it (you shouldn't be using these if you have no medical training, which fuels down to someone powergaming or not powergaming) and accessibility. 

The kits aren't that accessible to people on the ship if they don't have Medical access. Sure, they can loot the maint. shafts and the first aid cabinets, but they're limited in number and not everyone can loot the shafts, ICly and OOCly.

Now, specifically for the changes themselves. People rarely use gauze and ointment. People rarely carry kits with themselves, because they're not worth it on their own. These changes, in my eyes, change the dynamic this way:

- Advanced first aid kit is now fully brute damage focused, I sort of agree on the tramadol being abuseable, however having tramadol would be useful.
- Burn kit is actually useful now.
- Oxy kit seems fine. 
- Toxin kit is a lovely change. On a ship working with varied radioactive sources, it's insane that a toxin kit doesn't have hyronalin. And do take note, everyone, having a hyro pill bottle won't make someone instantly immune to rads. Hyronalin works slowly, and your liver, eyes and all other organs can get damaged in the process if you don't respond fast enough. The kit also isn't that accessible.
- The emergency oxy tank change is great. It makes no sense for the emergency oxy tanks to last for like,  a minute or so. In any situation that a regular civilian passenger Joe, despite getting a soft suit and all that jazz, they still end up dying due to lack of access, tools and most importantly OXYGEN. This at least helps them get help in time. IMO, I would have the tanks start fully filled up.

Overall, I find the kits to be a change in pace. Yes, they can be abused with these changes. However, it helps medical contractors, corpsmen and non-Chem trained personnel get actual medicine and get people stabilized. This does not instantly treat people. They will still be hurt and fucked up. They will still need a checkup. They will still need surgery. This just makes it less tedious for people to get the tools necessary to do it. 


But in the end, if this passes, some people will always abuse the tools, so, whatever.

Spookerton commented:
I totally get the thought process behind these changes, but they're fundamentally bad for gameplay, bad for balance, and bad for medical - both as a system and as a job group. They make access to anything you could possibly need except surgery completely trivial, severely reduce almost all danger in almost all ways, and encourage powergaming.

Do not want. Bad. No.

thefrostycoder commented:
Personally, I think first-aid in general needs a rework. Because some of these points are right- as it stands, the kits that usually spawn in medbay aren't being used because most doctors already have the meds they need. Alternatively they aren't being used because no one can get at them. So. I think first aid kits in general should look something like: 

Really given the dangers that the torch faces, we only really need a handful of general first aid supplies. All kits should have bandages/gauze and splints, but additionally:

-  General First Aid (These should be scattered throughout the station in public areas
   - Autoinjector
   - Bottle of Paracetamol
   - Bottle of Antidexafen
   - Ointment

- Radiation Kit (Should be found mostly in Engineering, particularly near the engine monitoring area and the rust monitoring room)
   - Hyronalin Pills
   - Dermaline Pills
   - Maybe Leporazine pills?

- Burn Kit
   - Probably the usual burn kit.


At no point should someone who isn't medically trained be administering drugs via syringe or IV, in my opinion. At best if it's something that doesn't work as a pill it should be added in as an autoinjector. 

Additionally we might benefit from some clear distinctions on what qualifies as First Aid vs Medical Treatment

FreePrivateer commented:
Well, this is interesting.

I can see the thought process behind this as well.

Part of me thinks this is a good change. As mentioned by some other folks, at the moment, kits are kinda just... raided for stuff, if that. I'd argue that first aid kits can have two sorts of levels: Kits that are, presumably, supposed to be used by medically trained personnel, and kits that are supposed to be used by the rest of folks. An advanced kit in the engineering first aid locker isn't necessarily supposed to be used by engineers themselves, but might be something that a paramedic yells for someone to get, or that an engineer shakily uses while being instructed by medical over the radio, as far as I see it.

As the skills are written now, every crew member should be able to use ointment, bandages, and autoinjectors; I'd argue that 'untrained' personnel would also be able to take pills, but won't necessarily know what all those pills do, how many they should take, if they should take them, etc. Untrained personnel shouldn't 'know' how to use advanced burn or trauma kits, either. 

The presumption of the objection here is that people won't follow their skill limitations, and use medications they can't identify, or tools they have no training to use. 

This change makes /most/ medical kits the type that someone /with/ training can snatch and use, but additionally something that the untrained crew might not be able to use. 

I'd rather the autoinjectors sit at 10 units, rather then 5, so they're actually useful.

Keep the basic first aid kit as something untrained crew can use; fill it with two bruise packs, two ointments, a bottle of cold meds, a bottle of weak painkillers, two inaprov auto injectors. Something someone can use if they zapped themselves on a door, burned themselves on a lighter, cut themselves with a knife by accident, got the sniffles, dropped a toolbox on their foot; Should someone look in a bad way, it has the tool's for Crewman W.T. Door to first aid someone with no training. Rename the pill bottles to 'cold medicine' and 'painkillers' to emphasize that untrained personnel can use them. Then, remap those to most of the placed where a first aid kit might be.

Put more specialized kits where they might be appropriate, make them a bit rarer; Those kits are for nurses or departmental crew that are trained as EMTs.

Keep the burn kit and the trauma kit in the medbay for medics and doctors to raid.

afterthought2 commented:
I think that at least some of these ideas have merit.

-Please don't change the advanced kits. These are not meant for trauma specifically, they are meant for med players generally. They are by far the most used kits.
-Adding tramadol to the burn kits is probably OP. Adding extra burn kits or at least ointment would be OK. Maybe paracetamol autoinjectors.
-Adding hyronalin to the toxin kits is at least interesting. I feel that somewhere, either in the kits or elsewhere in the infirmary, there should be some hyronalin pills. Corpsmen and medical contractors don't get chemistry access, so if there is no physician (or the elusive chemist) you're out of luck. This is pretty new, since the chemist role change, and I think the kit loadouts don't account for it properly yet.
-The oxygen kits are kind of stupid. Dexaline is not a useful drug unless the lungs are extremely damaged, in which case it's still borderline. An obvious way to improve them is to include an oxygen tank. But wait--that's the crew survival kit. The only thing I can think of is to add iron/sugar pills, as the most common reason for oxygen deprivation which is kit-treatable (lung/heart damage is not) is blood loss, and these pills are relatively effective. As a balancing mechanism, you could remove the health analyzer that I think this kit comes with (unlike any of the others) so it won't be so obvious how many pills you have to take/whether you still need to go to the infirmary if you are self-treating.
-Oxygen tank level: please do this. I don't understand why they have such low pressure now: surely you would carry a filled oxygen tank, not an almost empty one, around?

Something which is not mentioned but could be used as a balancing mechanism is how full the pill boxes are, and how big the pill is. After the pill bottle size increase (which was great), the default bottles start with a lot of pills. Maybe 21 15u hyronalin pills is OP, but 6 5u hyronalin pills is enough for one, maybe two, people, and they each have to take multiple.

In any case, I do think this is an area that would benefit from some adjustments, especially in light of the recent medical changes. Maybe what's proposed here right now makes kits too strong, but it's worth discussing.

Novacat commented:
I did some changes based off of feedback. Ideally, one of the problems I am facing is that painkillers in the game are either too useful in combat, or too weak to combat pain and stop pain shock. For now, though, I will be removing all tramadol except for a few autoinjectors.

Though, perhaps Soporifics? Can soporifics stop pain shock?

Advanced Kits: Changed them to the way they used to be, but swapped an inaprovaline autoinjector for a tramadol autoinjector.

Fire kits: Swapped burn kits for ointment, removed a tramadol autoinjector

Oxygen: Buffed Dexalin to be less useless, reduced the number of autoinjectors. Will probably throw in sugar-iron pills later.

Toxin: Reduced the number of autoinjectors, reduced the number and dosage of hyronlin pills.

Novacat commented:
Just added soporific pills and sugar-iron pills, as suggested. Sugar-iron pills have been added to the oxygen kit, and soporific pills replaces paracemetol in the burn kit. Very slightly increased the effects of paracemetol.

ghost commented:
Soporifics are relaxants (the chemical itself is literally named "sleep toxin" in the code), and ultimately make you sleep. Why would you put them in medkits, and for what purpose?

Rebruiza commented on code/game/objects/items/weapons/storage/firstaid.dm:
```diff
@@ -252,3 +254,21 @@
+
+/obj/item/weapon/storage/pill_bottle/sugariron
+	name = "bottle of Sugar-Iron pills"
+	desc = "Used to help the body naturally replenish blood.."
```
Double punctuation.

Rebruiza commented on code/modules/organs/blood.dm:
```diff
@@ -304,9 +304,9 @@ proc/blood_splatter(var/target,var/datum/reagent/blood/source,var/large,var/spra
 	var/oxygenated_mult = 0
 	if(chem_effects[CE_OXYGENATED] == 1) // Dexalin.
-		oxygenated_mult = 0.5
+		oxygenated_mult = 0.7
```
This almost puts Dexalin in line with Dexalin Plus. Why?

Rebruiza commented on code/modules/reagents/reagent_containers/hypospray.dm:
```diff
@@ -116,6 +116,8 @@
 	volume = 5
 	origin_tech = list(TECH_MATERIAL = 2, TECH_BIO = 2)
+	slot_flags = SLOT_BELT | SLOT_EARS
+	w_class = ITEM_SIZE_TINY
```
Doesn't **ITEM_SIZE_TINY** allow you to put it behind the ears?

Novacat replied:
I based it off syringe code, which has the same.

Rebruiza replied:
Hypospray =/= Syringe

EcklesFire commented:
14 sugar-iron pills in a box is overpowered as heck, you can sit there and replenish your blood loss easily with that.

Not a fan at all, they're crazy good for bringing yourself back from the brink.

ghost commented:
I am not sure why this PR aims to move the medbay to medkits, when the medbay is literally created for curing what you can't cure out in the field. Medkits should be for stabilization, not treating you 100%.

ghost commented:
Also, you should probably squash your commits.

atlantiscze commented on code/game/objects/items/weapons/tanks/tank_types.dm:
```diff
@@ -92,14 +92,14 @@
 	force = 4
 	distribute_pressure = ONE_ATMOSPHERE*O2STANDARD
-	volume = 2 //Tiny. Real life equivalents only have 21 breaths of oxygen in them. They're EMERGENCY tanks anyway -errorage (dangercon 2011)
+	volume = 3 //Tiny. Real life equivalents only have 21 breaths of oxygen in them. They're EMERGENCY tanks anyway -errorage (dangercon 2011)
```
Rather than increasing volume of the tank (which increases its maximal capacity as well), i recommend keeping only changes to starting_pressure.

If anything, increase starting_pressure to 10 atmospheres (maximal value when filling from a canister) to cut on the busywork of filling up your emergency tank every round (it also makes sense for them to start filled).

This keeps the maximal capacity unchanged, but solves the problem (tanks as you find them last short time).

Emergency oxygen tank capacities feel to me as well balanced at this time, the only black spot here is the large oxygen tank, which lasts for something around 12 hours of real time now, which effectively means that you don't have to care about oxygen for the rest of the round if you have one of these.

afterthought2 commented:
Buffing dexalin to 70% is quite strong. If you do it, I don't think you should give the iron pills as well, and it does now overlap more with dexalin+. I actually think that's OK, but you may get pushback from others. Right now, I find I use dex+ in three cases: reoxygenation after lungs transplant, prior to/awaiting lung surgery, and to stabilize people with bad lungs or oxygen-deprived atmosphere when access to surgery is impossible for a longish time (e.g. Charon). Compared to this, the old dexalin is only partly useful in the first role, and useless in the others. The new dexalin would be useful in all of these roles, though still less effective. The difference between 70 and 80 is actually much larger than it seems: the relevant thresholds for taking brain damage are 85, 75, 60, and 40. So 70 means you are vulnerable to more damage than 80 (into the region untreatable with inaprovaline anymore), and will start getting the blurred vision/paralysis effect. Keep in mind that this stacks with blood loss, so often the actual oxygenation will be lower.

Buffing dex+ to 85% is uncalled for, in my opinion.

Rebruiza and some others above are concerned about having medkits put medical staff out of work. I think the buff to dexalin is therefore good precisely because it doesn't do that: dexalin does not fix any of your problems, it just buys time to get to the infirmary.

Novacat committed:
First Aid Changes


Emergency Tank Tweaks


Reverted autoinjectors to 5u


Feedback Changes


Feedback Changes 2


Fixed an error


Feedback Changes 3


Feedback Changes 4


Grammar


Merge remote-tracking branch 'origin/MiscTweaks' into MiscTweaks
Grammar

Novacat committed:
Reduced Dexalin Plus

Novacat commented:
Reduced Dexalin Plus to 80% and Dexalin to 60%, reduced the number of sugar-iron pills to just 2 in the oxy kit, and reverted soporifics. I kept the soporific and sugar iron pill bottles in case another dev wishes to use them.

Rebruiza commented on code/game/objects/items/weapons/tanks/tank_types.dm:
```diff
@@ -117,7 +117,7 @@
 	icon_state = "emergency_nitro"
 	gauge_icon = "indicator_emergency"
-	starting_pressure = list("nitrogen" = 3*ONE_ATMOSPHERE)
+	starting_pressure = list("nitrogen" = 10*ONE_ATMOSPHERE)
```
This has more than the bigger variant?

Rebruiza commented on code/game/objects/items/weapons/tanks/tank_types.dm:
```diff
@@ -133,4 +133,4 @@
 	distribute_pressure = ONE_ATMOSPHERE*O2STANDARD
-	starting_pressure = list("nitrogen" = 3*ONE_ATMOSPHERE)
\ No newline at end of file
+	starting_pressure = list("nitrogen" = 6*ONE_ATMOSPHERE)
```
Why?

Novacat replied:
Brings it up to oxygen tanks, which are already at 6 atmospheres.

Rebruiza replied:
:+1:

ghost commented:
>I kept the soporific and sugar iron pill bottles in case another dev wishes to use them.

Then they can add the relevant code themselves when "they wish to use it". No need for useless code bloat if it's not going to be used anyways.

Novacat committed:
Feedback Changes 5

Novacat commented:
Advanced Med kits now get an assorted pill bottle instead of the autoinjector, which has an assortment of critical medications. 

Removed extraneous pill bottles.

Removed sugariron pills from oxy kit.

Reset Dexalin/Dexalin Plus, since my changes were mostly due to a misinterpretation of how they worked.

ghost commented:
I actually like how this makes it so you can ignore chemistry, but chemistry is still needed if you want everything to function smoothly.

(By that, the most basic medications are available so you can survive a round without a chemist, but a chemist is still needed if you want even more effective medication, and some medication that is more of a niche.)

PurpleMartinJCK reviewed (dismissed):
You've been working really well with the dev team on this PR which is awesome, thank you for that @Novacat. I think this is looking pretty good now actually. You've improved the usability of these kits without making them OP.

comma commented on code/game/objects/items/weapons/storage/firstaid.dm:
```diff
@@ -57,8 +57,9 @@
-		/obj/item/weapon/reagent_containers/syringe/antitoxin = 3,
+		/obj/item/weapon/reagent_containers/hypospray/autoinjector/detox = 4,
 		/obj/item/weapon/storage/pill_bottle/antitox,
+		/obj/item/weapon/storage/pill_bottle/hyronalin,
```
not really digging entire pill bottle of this, just give them a single loose pill.
it's first aid box, not 'heal any radiation effects' box.

comma commented on code/game/objects/items/weapons/storage/firstaid.dm:
```diff
@@ -86,7 +87,7 @@
 	item_state = "firstaid-advanced"
 
 	startswith = list(
-		/obj/item/weapon/reagent_containers/hypospray/autoinjector,
```
IMO injector is still important to have, so if it still fits stick it back

This kit is kinda weird because its not really used as first aid, its more of general med supplies storage.

Novacat replied:
It does not fit unfortunately.

Novacat committed:
Feedback Changes 6
